From 8aa117af975612cca32ef361dac4bcaebfe9e8e9 Mon Sep 17 00:00:00 2001
From: Florian Reichl <63750742+reichlfl@users.noreply.github.com>
Date: Sat, 15 Jun 2024 19:28:04 +0200
Subject: [PATCH 03/14] [BACKEND] Update LLVM version to
 https://github.com/llvm/llvm-project/commit/e4790ce (#4106) (#4107)

https://github.com/llvm/llvm-project/commit/e4790ce2096c805b7598c353f9e7fa513701f843

Rationale for the code changes: Since
https://github.com/llvm/llvm-project/commit/4bce270157f9a81bd7e38dc589a2970a445d1e9,
`value` may be an `UndefAttr`, but `SplatElementsAttr` expects a
`FloatAttr` or `IntegerAttr`.
---
 lib/Dialect/Triton/IR/Ops.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/Dialect/Triton/IR/Ops.cpp b/lib/Dialect/Triton/IR/Ops.cpp
index ce4f97336..62e8667f8 100644
--- a/lib/Dialect/Triton/IR/Ops.cpp
+++ b/lib/Dialect/Triton/IR/Ops.cpp
@@ -543,6 +543,8 @@ OpFoldResult SplatOp::fold(FoldAdaptor adaptor) {
   auto value = adaptor.getSrc();
   if (!value)
     return {};
+  if (!isa<FloatAttr, IntegerAttr>(value))
+    return {};
   auto shapedType = cast<ShapedType>(getType());
   auto ret = SplatElementsAttr::get(shapedType, ArrayRef<Attribute>(value));
   return ret;
